### PR TITLE
Feature/add sights filters

### PR DESCRIPTION
**Pull request by rohitsharma120582**
This PR is about adding Category filter and Search for sights data...

Tasks:
- [x] add UI for Category filter and Search with sticky position
- [x] make category filters list dynamic
- [x] implement search functionality with debouncing
- [x] implement filter functionality on category property
- [x] load sights data from relative `./packages/sights` instead of actual NPM package

### CODE REVIEW

File: website/src/pages/Sights/index.js [hunk 1]
```diff
@@ -1,91 +1,86 @@
...
 
   const darkTheme = createTheme({
     palette: {
       mode: 'dark',
     },
   });
 
+  useEffect(() => {
+    const filteredSights = Object.values(sightsData)
+      .filter((sight) => (category === '' || category === sight.category))
```
Comment (souyahia-monk) [inline website/src/pages/Sights/index.js:91]: if the sight does not have any category why should it be included in the category filter ?

File: website/src/pages/Sights/index.js [hunk 2]
```diff
@@ -1,91 +1,86 @@
...
+      .filter((sight) => {
+        const value = JSON.stringify(sight).toLocaleLowerCase();
+        return typeof value === 'string' && value.includes(search.toLocaleLowerCase());
```
Comment (souyahia-monk) [inline website/src/pages/Sights/index.js:94]: why are you checking if the return value is a string ?

File: website/src/pages/Sights/index.js [hunk 3]
```diff
@@ -1,91 +1,86 @@
...
+      });
+    setSights(filteredSights);
+  }, [search, category]);
+
+  useEffect(() => {
+    const categoryArr = sights.map((sight) => sight.category);
+    const categories = categoryArr.filter((item, index) => categoryArr.indexOf(item) === index);
```
Comment (souyahia-monk) [inline website/src/pages/Sights/index.js:101]: I think : 

```javascript
const categories = sights.map((sight) => sight.category)
  .filter((item, index) => categoryArr.indexOf(item) === index);
````

is simpler

File: website/src/pages/Sights/index.js (hunk 1 above)
Comment (rohitsharma120582) [reply website/src/pages/Sights/index.js:91]: Sorry, I didn't get your point. But, according to this logic:
- it should return every sight, if no category selected by user
- it should return every sights which has the same category what user has selected

Is there anything wrong here?

File: website/src/pages/Sights/SightCard/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        <Typography
+          gutterBottom
+          variant="h6"
+          component="div"
+          sx={{
+            display: 'flex',
+            justifyContent: 'space-between',
+          }}
+        >
+          {/* eslint-disable-next-line react/prop-types */}
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply website/src/pages/Sights/SightCard/index.js]: Sorry, it was already there and I didn't check it on commit. I will fix it.

File: website/src/pages/Sights/index.js (hunk 3 above)
Comment (rohitsharma120582) [reply website/src/pages/Sights/index.js:101]: We require `categoryArr` which will be missing if merge this code that's why I have broken this code in 2 lines.

File: website/src/pages/Sights/index.js (hunk 2 above)
Comment (rohitsharma120582) [reply website/src/pages/Sights/index.js:94]: Yeah! I will fix it.

File: website/src/pages/Sights/SightCard/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply website/src/pages/Sights/SightCard/index.js]: Fixed.

File: website/src/pages/Sights/index.js (hunk 2 above)
Comment (rohitsharma120582) [reply website/src/pages/Sights/index.js]: Fixed.

**Pull request merged**